src/lib/pubkey/ec_group/ec_group.h

```diff
@@ -43,6 +43,21 @@ enum class EC_Group_Source {
    ExternalSource,
 };
 
+/**
+* Enum indicating the way the group in question is implemented
+*
+* This is returned by EC_Group::engine
+*/
+enum class EC_Group_Engine {
+   /// Using per curve implementation; fastest available
+   Pcurve,
+   /// A generic implementation that handles many curves in one implementation
+   PcurveGeneric,
+   /// The old implementation, used as a fallback
+   /// TODO(Botan4) remove this
+   LegacyBigInt,
+};
```

Comment:
One consideration from reading the docstrings: is "pcurve" actual lingo that we want to expose to "casual" users?

Perhaps:

```suggestion
/**
* Enum indicating the way the group in question is implemented
*
* This is returned by EC_Group::engine
*/
enum class EC_Group_Engine {
   /// Using per curve implementation; fastest available
   Optimized,
   /// A generic implementation that handles many curves in one implementation
   Generic,
   /// The old implementation, used as a fallback
   /// TODO(Botan4) remove this
   Legacy,
};
```